veps/sig-compute/104_allow_disabling_feature_gates/vep.md

```diff
@@ -0,0 +1,253 @@
+    featureGates:
+      - name: VMExport
+        enabled: true
+      - name: ImageVolume
+        enabled: false
+      - name: DownwardMetrics
+        # defaults to true
+    developerConfiguration:
+      featureGates:
+        - IncrementalBackup # This enables the FG
+        - ImageVolume # Being ignored since the same FG is listed in the above config which takes precedence
+```
+
+Pros:
+* Extensible: a struct opens the door for further configuration moving forward.
+* Aligns with Kubernetes best practices: "the convention is to use a list of subobjects containing name fields ... 
+This rule maintains the invariant that all JSON/YAML keys are fields in API objects".
+See [Kubernetes api conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#lists-of-named-subobjects-preferred-over-maps)
+for more info.
+
+Cons:
+* Verbose: when there is no desire to disable feature gates, the admin would now have to supply the `name: ` boilerplate
+which makes the manifest more verbose.
+
+### Approach #2 - Use a new map structure from a string to bool
+
+As part of this approach, a new `kv.spec.configuration.featureGateMap` will be added of type `map[string]bool`.
+That is a map from string to bool types, i.e. feature name to whether it's enabled or not.
+
+`kv.spec.configuration.featureGates` has precedence over `kv.spec.configuration.developerConfiguration.featureGates`.
+However, if a feature gate exists only in the legacy feature gate slice, it would be considered enabled.
+This way we keep backward compatibility.
+
+Usage Example:
+```go
+kind: KubeVirt
+spec:
+  configuration:
+    featureGateMap:
+      VMExport: true
+      ImageVolume: false
+      DownwardMetrics: false
+    developerConfiguration:
+      featureGates:
+        - IncrementalBackup # This enables the FG
+        - ImageVolume # Being ignored since the same FG is listed in the above config which takes precedence
+```
+
+Pros:
+* Simplicity (?): Easy to implement and understand.
+
+Cons:
+* Goes against Kubernetes best practices: "the convention is to use a list of subobjects containing name fields".
+The api conventions specifically mention using maps as an anti-pattern:
+"Lists of named subobjects preferred over maps ...  There are no maps of subobjects in any API objects".
+See [Kubernetes api conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#lists-of-named-subobjects-preferred-over-maps)
+for more info.
+* Maps are, by definition, unordered. This means that users/clients always have to sort it in a consistent manner
+to avoid entities mistakenly assuming that the state had changed because a change of ordering.
+* Non-extensible: If we'll need to add more configurations to feature gates in the future, we'll be in the same problem
+of having to extend the API.
+
+### Approach #3 - Use a complementary slice for disabled feature gates
```

Comment:
From my point of view, this is the only solution that preserves "comparability" as elaborated in the 3rd comment at https://github.com/kubevirt/enhancements/pull/105/files#r2596246960 .

It may look less attractive, but it seems the safest and most compatible solution, without adding logic to sync two inputs.